src/ifaddrs.rs

```diff
@@ -0,0 +1,180 @@
+pub struct InterfaceAddress {
+    /// name of the interface
+    pub interface_name: String,
+    /// flags from `SIOCGIFFLAGS`
+    pub flags: InterfaceFlags,
+    /// address of interface
+    pub address: Option<InetAddr>,
+    /// netmask of interface
+    pub netmask: Option<InetAddr>,
+    /// broadcast address of interface, if applicable
+    pub broadcast: Option<InetAddr>,
+    /// point-to-point destination address
+    pub destination: Option<InetAddr>,
+}
+
+impl InterfaceAddress {
+    /// Create an `InterfaceAddress` from the libc struct.
+    fn from_libc_ifaddrs(info: &libc::ifaddrs) -> InterfaceAddress {
+        let ifname = unsafe { ffi::CStr::from_ptr(info.ifa_name) };
+        let address = unsafe { InetAddr::from_libc_sockaddr(info.ifa_addr) };
+        let netmask = unsafe { InetAddr::from_libc_sockaddr(info.ifa_netmask) };
+        let mut addr = InterfaceAddress {
+            interface_name: ifname.to_string_lossy().to_string(),
+            flags: InterfaceFlags::from_bits_truncate(info.ifa_flags as i32),
+            address: address,
+            netmask: netmask,
+            broadcast: None,
+            destination: None,
+        };
+
+        #[cfg(any(target_os = "linux", target_os = "emscripten", target_os = "fuchsia"))]
+        {
+            let ifo = unsafe { InetAddr::from_libc_sockaddr(info.ifa_ifu) };
+            if addr.flags.contains(IFF_POINTOPOINT) {
+                addr.destination = ifo;
+            } else if addr.flags.contains(IFF_BROADCAST) {
+                addr.broadcast = ifo;
+            }
+        }
+
+        #[cfg(any(target_os = "macos", target_os = "ios", target_os = "freebsd",
+                    target_os = "dragonfly", target_os = "openbsd", target_os = "netbsd",
+                    target_os = "bitrig", target_os = "solaris"))]
+        {
+            addr.destination = unsafe { InetAddr::from_libc_sockaddr(info.ifa_dstaddr) };
+        }
+
+        addr
+    }
+}
+
+/// Holds the results of `getifaddrs`.
+///
+/// Use the function `getifaddrs` to create this Iterator. Note that the
+/// actual list of interfaces can be iterated once and will be freed as
+/// soon as the Iterator goes out of scope.
+pub struct InterfaceAddressIterator {
+    base: *mut libc::ifaddrs,
+    next: *mut libc::ifaddrs,
+}
+
+impl Drop for InterfaceAddressIterator {
+    fn drop(&mut self) {
+        unsafe { libc::freeifaddrs(self.base) };
+    }
+}
+
+impl Iterator for InterfaceAddressIterator {
+    type Item = InterfaceAddress;
+    fn next(&mut self) -> Option<<Self as Iterator>::Item> {
+        match unsafe { self.next.as_ref() } {
+            Some(ifaddr) => {
+                self.next = ifaddr.ifa_next;
+                Some(InterfaceAddress::from_libc_ifaddrs(ifaddr))
+            }
+            None => None,
+        }
+    }
+}
+
+/// Get interface addresses using libc's `getifaddrs`
+///
+/// Note that the underlying implementation differs between implementations and
```

Comment:
"the underlying implementation differs between implementations", that's pretty redundant. Can you rephrase this to be a little more clear?